cloud/azure/README.md

```diff
@@ -1,7 +1,85 @@
-# Azure Terraform 
+# Azure Terraform Setup
+We use a remote backend to store state information to avoid conflicts when 
+multiple collaborators are updating resources managed by Terraform. 
+Follow the steps outlined in this doc to enable collaborators to access 
+the remote backend and shared state on their machines.
 
-## Overview
-Currently the `main.tf` file includes the main terraform configuration file that sets up the different services that civiform uses in Azure. 
+# Setup
+In order to run for the first time run
+ 
+```
+$ source setup
+```
 
-## Variables
-`main.tf` uses variables via `var.var_name`. Those variable definitions and defaults live in `variables.tf`. For values that need to be private create a `.auto.tfvars` or `terraform.tfvars` file which should not be checked into a public repo (as it will often contain private information). We have added an example file to copy and fill out with specific to deployment infromation
+This will: 
+1. set up a resource group for terraform shared state to live in if it does
+not already exist
+2. make a blob storage account for the shared tfstate to live in if it does
+not already exist
+3. create a backend_vars file that will be used when running terraform to 
+share the state (will overwrite the existing)
+
+Note that it takes azure a few minutes for you to be able to access the blob 
+storage and that you'll need to share the backend_vars with other people 
+developing on your project. You can find the info on the azure portal as well. 
+
+You can run this command again if you delete your backend_vars or need to
+re-export your account key (although you can also re-export your account key
+via the commands below), note that it does a pretty fuzzy regex match for
+the storage account name so if you are doing something special you might have
+to wright your own script to do that!
+
+## Export your account key
+In order to use terraform with the azure backend you'll need to get the 
+`account_key` set up properly. The backend_vars file expects it to be provided 
+via ARM_ACCESS_KEY so you'll need to use the Azure CLI to provide it. 
+Get the `resource_group_name` and `storage_account_name` from the backend_vars
+file or via the azure cli/portal (details on how to do this in the 
+troubleshooting section below)
+```
+ACCOUNT_KEY=$(az storage account keys list --resource-group \
+$RESOURCE_GROUP_NAME --account-name $STORAGE_ACCOUNT_NAME --query \
+'[0].value' -o tsv)
+export ARM_ACCESS_KEY=$ACCOUNT_KEY
+```
+
+## Running terraform with this config  
+Since we are using a shared backend you'll need to run init specifying that
+```
+$ terraform init -backend-config=backend_vars
```

Comment:
question (blocking): please mention which pwd the reader should use when running these commands